.github/workflows/ubuntu.yml

```
@@ -78,6 +78,42 @@ jobs:
         python3 -m pip install --upgrade pip setuptools wheel
         PYWARPX_LIB_DIR=$PWD/build_sp/lib python3 -m pip wheel Python/
 
+  build_icpx:
+    name: oneAPI ICX SP [Linux]
+    runs-on: ubuntu-20.04
+    steps:
+    - uses: actions/checkout@v2
+    - name: install dependencies
+      shell: bash
+      run: |
+        .github/workflows/dependencies/dpcpp.sh
+    - name: build WarpX
+      shell: bash
+      run: |
+        set +e
+        source /opt/intel/oneapi/setvars.sh
+        set -e
+        export CXX=$(which icpx)
+        export CC=$(which icx)
+
+        cmake -S . -B build_sp         \
+          -DCMAKE_CXX_COMPILER_ID="Clang"                \
```

More work-arounds for broken Intel patch that landed in oneAPI:
```suggestion
          -DCMAKE_C_COMPILER_ID="Clang"                  \
          -DCMAKE_CXX_COMPILER_ID="Clang"                \
```